internal/modelcontroller/pod_utils.go

```diff
@@ -0,0 +1,65 @@
+package modelcontroller
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+func (r *ModelReconciler) execPod(ctx context.Context, pod *corev1.Pod, container string, command []string) error {
+	execReq := r.PodRESTClient.
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: container,
+			Command:   command,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+		}, runtime.NewParameterCodec(r.Scheme))
+
+	exec, err := remotecommand.NewSPDYExecutor(r.RESTConfig, "POST", execReq.URL())
+	if err != nil {
+		return fmt.Errorf("creating remote command executor: %w", err)
+	}
+
+	if err := exec.Stream(remotecommand.StreamOptions{
```

Comment:
not sure if this is important or not:

exec.Stream is deprecated: use StreamWithContext instead to avoid possible resource leaks. See https://github.com/kubernetes/kubernetes/pull/103177 for details

VS code complained about it.

Reply:
That does sound like it would be good to do

Reply:
Updated